test/libraries/SafeToL2Setup.spec.ts

```diff
@@ -0,0 +1,206 @@
+                // To do that, we search for the DELEGATECALL opcode with the stack input pointing to the library address
+                const delegateCallIntoTheLib = trace.structLogs.findIndex(
+                    (log) =>
+                        log.op === "DELEGATECALL" &&
+                        sameString(log.stack[log.stack.length - 2], alignHexString(safeToL2SetupLibAddress, 32)),
+                );
+                const preDelegateCallStorage = trace.structLogs[delegateCallIntoTheLib].storage;
+
+                // The SafeSetup event is emitted after the Safe is set up
+                // To get the storage snapshot after the Safe is set up, we need to find the LOG2 opcode with the topic input on the stack equal the SafeSetup event signature
+                const SAFE_SETUP_EVENT_SIGNATURE = ethers.keccak256(
+                    ethers.toUtf8Bytes("SafeSetup(address,address[],uint256,address,address)"),
+                );
+                const postSafeSetup = trace.structLogs.find(
+                    (log, index) =>
+                        log.op === "LOG2" &&
+                        log.stack[log.stack.length - 3] === SAFE_SETUP_EVENT_SIGNATURE.slice(2) &&
+                        index > delegateCallIntoTheLib,
+                );
+                if (!postSafeSetup) {
+                    throw new Error("No SafeSetup event");
+                }
+                const postSafeSetupStorage = postSafeSetup?.storage;
+
+                for (const [key, value] of Object.entries(postSafeSetupStorage)) {
+                    // The slot key 0 is the singleton storage slot, it must equal the L2 singleton address
+                    if (key === alignHexString("00", 32)) {
+                        expect(sameString(alignHexString(safeL2SingletonAddress, 32), value)).to.be.true;
+                    } else {
+                        // All other storage slots must be the same as before the DELEGATECALL
+                        if (key in preDelegateCallStorage) {
+                            expect(sameString(preDelegateCallStorage[key], value)).to.be.true;
+                        } else {
+                            // This special case is needed because the SafeToL2Setup library inherits the SafeStorage library
+                            // And that makes the tracer report all the storage slots in the SafeStorage library as well
+                            // Even though if they were not touched during the transaction
+                            expect(sameString(value, "0".repeat(64))).to.be.true;
+                        }
+                    }
+                }
+
+                // Double-check that the storage slot was changed at the end of the transaction
+                const singletonInStorage = await hre.network.provider.send("eth_getStorageAt", [safeAddress, `0x${"0".repeat(64)}`]);
```

Comment:
ubernit: The same value is constructed in different ways which annoys my OCD; Here its `0x${"0".repeat(64)}`, there its `alignHexString("00", 32)`.